### PR TITLE
DRM driver cleanups

### PR DESCRIPTION
**Pull request by JoseExposito**
Hi!

I recently bought a StarFive VisionFive board and, since I'm trying to learn more about the DRM subsystem, I'm looking into the KMS driver code.

I don't know what's the status of the driver, if there's a list of pending features you'd like to implement or if you'd be interesting in sending the code to the staging tree and work upstream... So, this PR is my way of saying hello and get more information about the project, as well as a way to contribute some trivial patches.

Looking forward to your response,
Jose

### CODE REVIEW

**Comment by JoseExposito** [comment]
Sorry, I forgot to ping @esmil as noted in the README. Let me know if you want me to rebase my branch.

File: drivers/gpu/drm/starfive/starfive_drm_drv.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 struct starfive_drm_private {
 	struct drm_fb_helper fbdev_helper;
 	struct drm_gem_object *fbdev_bo;
+	/* Protects mm. */
```
Comment (esmil) [inline drivers/gpu/drm/starfive/starfive_drm_drv.h:17]: This comment is not really helpful. What exactly is the "mm" that it is protecting? The drm_mm structure below?

**Review by esmil — COMMENTED** [review]
Hi Jose,

I've now squashed the whitespace and typo fixes into the commit adding the drm driver. I'm not sure about deleting the commented out code. The whole driver is very much a half-finished work, so it might be good to have a comment reminding about missing pm ops fx.

File: drivers/gpu/drm/starfive/starfive_drm_drv.h (hunk 1 above)
Comment (JoseExposito) [reply drivers/gpu/drm/starfive/starfive_drm_drv.h:17]: `mm` is the name of the variable this mutex should protect... But I just noticed that both `mm_lock` and `mm` are not used, so we should probably remove them.

**Comment by JoseExposito** [comment]
Hi @esmil,

Thanks a lot for reviewing and taking my patches.

> I'm not sure about deleting the commented out code. The whole driver is very much a half-finished work, so it might be good to have a comment reminding about missing pm ops fx.

Speaking about unfinished work, is there a TODO list of features you know are pending? There is still clean up work pending and some APIs are deprecated and need to be updated, but I'm not sure if you know something else that is pending.

Also, do you have plans to upstream in the staging tree this driver? It'd make easier working on it following the usual mail workflow.

Closing this PR, I'd try to open a new one with more fixes. Thanks!!

**Pull request closed without merging**